### PR TITLE
algolia improvements

### PR DESCRIPTION
**Pull request by nsollecito**
### What does this PR do?

For algolia and search
- Adds some better code snippet highlights where possible
- Breaks down large pieces of content under `<h2>` into smaller chunks in the index. Previously some of this was being truncated.
- Adds some small breadcrumb to the dropdown results to better indicate where you are navigating.

### Motivation

Search improvements

### Preview

https://docs-staging.datadoghq.com/nsollecito/algolia-full-pages/

- Try searching for `aws.lambda.provisioned_concurrency_invocations` previously this metric was being truncated and wasn't returned in the search.  Another good test for this is the vsphere integration which has a lot of integrations, try searching for `datadog.vsphere.refresh_metrics_metadata_cache.time`
- Try searching for `agent install` and see if the results make sense
- Try searching for `eks logging` and see if the results make sense
- Try searching for `unified service tagging` and see if the results make sense
- Try searching other keywords you choose and see if the results make sense

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: assets/scripts/config/config-docs.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ module.exports = {
         ddApplicationId: '3493b4e7-ab12-4852-8836-ba96af7bc745',
         loggingHandler: 'http',
         algoliaConfig: {
-            index: 'docs_preview',
+            index: 'docs_preview_nick',
```
Comment (davidejones) [inline assets/scripts/config/config-docs.js:20]:
```suggestion
            index: 'docs_preview',
```
Apply this before merging

File: local/bin/js/algolia-index-sync.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const getIndexName = () => {
             algoliaIndexName = configDocs['live'].algoliaConfig.index;
             break;
         case 'preview':
-            algoliaIndexName = configDocs['preview'].algoliaConfig.index;
+            algoliaIndexName = 'docs_preview_nick';
```
Comment (davidejones) [inline local/bin/js/algolia-index-sync.js:13]:
```suggestion
            algoliaIndexName = configDocs['preview'].algoliaConfig.index;
```
apply this before merging

**Review by kayayarai — APPROVED** [review]

File: assets/scripts/components/algolia/searchbarHits.js [hunk 3]
```diff
@@ -88,15 +88,17 @@ const renderHits = (renderOptions, isFirstRender) => {
         const joinedListItems = hitsArray
             .map((item) => {
                 const hit = getHitData(item);
-                const displayContent = truncateContent(hit.content, 145);
+                const displayContent = truncateContent(hit.content_snippet_match_level === "full" ? hit.content_snippet : hit.content, 145);
+                console.log(hit.content_snippet_match_level);
```
Comment (nsollecito) [inline assets/scripts/components/algolia/searchbarHits.js:92]: remove before merge

**Review by devindford — APPROVED** [review]
Working as expected after updating the preview index

**Pull request merged**